test/object-store/sync/flx_migration.cpp

```diff
@@ -265,24 +267,107 @@ TEST_CASE("Test client migration and rollback", "[flx][migration]") {
     {
         auto realm = Realm::get_shared_realm(config);
 
+        REQUIRE(!wait_for_upload(*realm));
         REQUIRE(!wait_for_download(*realm));
+
+        auto table = realm->read_group().get_table("class_Object");
+        CHECK(table->size() == 5);
+    }
+
+    // Roll back to PBS
+    trigger_server_migration(session.app_session(), false, logger_ptr);
+
+    {
+        auto realm = Realm::get_shared_realm(config);
+
         REQUIRE(!wait_for_upload(*realm));
+        REQUIRE(!wait_for_download(*realm));
 
         auto table = realm->read_group().get_table("class_Object");
         CHECK(table->size() == 5);
     }
+}
+
+TEST_CASE("Test client migration and rollback with recovery", "[flx][migration]") {
```

Comment:
Since the server migration to FLX takes a long time, we may want to combine the tests as much as possible when the other test cases are added.

Reply:
I agree. I will think of a way to refactor them since we'll have a bunch of them.